File: web/src/types/dispatch.ts
// Mirror of Rust dispatch types

export type DispatchPriority = 'critical' | 'high' | 'normal';

export type StopAction = 'pickup' | 'dropoff';

export interface RouteStop {
  station_id: number;
  action: StopAction;
  bike_count: number;
  load_after: number;
}

export interface VehicleRoute {
  vehicle_id: number;
  capacity: number;
  stops: RouteStop[];
  total_distance_meters: number;
  estimated_duration_minutes: number;
}

export interface DispatchPlan {
  id: string;
  generated_at: string;
  vehicle_routes: VehicleRoute[];
  total_bikes_moved: number;
}

export interface DispatchVehicle {
  id: number;
  capacity: number;
  current_position: number;
}

File: crates/bike-core/src/dispatch.rs
use serde::{Deserialize, Serialize};
use uuid::Uuid;

use crate::domain::StationId;

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum DispatchPriority {
    Critical,
    High,
    Normal,
}

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum StopAction {
    Pickup,
    Dropoff,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct RouteStop {
    pub station_id: StationId,
    pub action: StopAction,
    pub bike_count: u32,
    pub load_after: u32,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct VehicleRoute {
    pub vehicle_id: u32,
    pub capacity: u32,
    pub stops: Vec<RouteStop>,
    pub total_distance_meters: f64,
    pub estimated_duration_minutes: f64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DispatchPlan {
    pub id: Uuid,
    pub generated_at: chrono::DateTime<chrono::Utc>,
    pub vehicle_routes: Vec<VehicleRoute>,
    pub total_bikes_moved: u32,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DispatchVehicle {
    pub id: u32,
    pub capacity: u32,
    pub current_position: StationId,
}
